pkg/scheduler/apis/config/v1/defaults.go

```diff
@@ -120,11 +120,6 @@ func SetDefaults_KubeSchedulerConfiguration(obj *configv1.KubeSchedulerConfigura
 		setDefaults_KubeSchedulerProfile(prof)
 	}
 
-	if obj.PercentageOfNodesToScore == nil {
```

Comment:
I think we should revert this so that we keep backwards compatibility.
Even though the behavior is the same, the output configuration should also be the same.

Reply:
> the output configuration should also be the same.

Hrm... I interpret that this is fixing a tiny "bug": in the before, serialization would output `percentageOfNodesToScore: 0` if it's not specified; with the change, nil PercentageOfNodesToScore won't show up, so isn't the current behavior more reasonable to reflect the original user input, so more portable?

Reply:
How is this different from other defaults?

We are just trying to be consistent with our API conventions. We wouldn't change the defaults for REST APIs.

Reply:
Aldo, I think the consensus was if it's nil (not set),  we won't change/set the value but use the default percentage to calculate the number of feasible nodes instead.

Reply:
> How is this different from other defaults?

For other defaults, if it's a "pointer-for-versioned, struct-for-unversioned" pattern, we have no choices - the serialized output of a struct has to carry its default value. It's just here both versioned and unversioned fields are both pointers, so we have the option now, so why not keep the output intact (to user's input)?

Reply:
I agree with Wei. We use the default percentage if it's empty but we should keep the config/profile output intact.

Reply:
> which looks more reasonable to me.

If we were writing the API from scratch, this is what we probably should have done. However, once we publish an API and we set defaults, we generally don't change them.

This is coming from an API review session we had offline.

Reply:
Got it.  I'll make the change.

Reply:
Reverted the change for backward compatibility.

Reply:
> Hrm... I interpret that this is fixing a tiny "bug": in the before, serialization would output `percentageOfNodesToScore: 0` if it's not specified; with the change, nil PercentageOfNodesToScore won't show up, so isn't the current behavior more reasonable to reflect the original user input, so more portable?

Another question, if we found a bug in API, what we will do then?
- For alpha stage, can we fix this directly?
- For beta/ga stage, I think we should bump a new version to fix this, am I right?

Reply:
> If we were writing the API from scratch, this is what we probably should have done. However, once we publish an API and we set defaults, we generally don't change them.

ok, that's fair.

Reply:
> Another question, if we found a bug in API, what we will do then?

It depends on the kind of bug. If it's impossible to create an object because of validation, we might do the change, but we have to do it across 2 kubernetes versions.

If we had a typo on a field or something like that, yes, we would have to create a new api version (always, regardless of alpha/beta/ga)